### ISSUE TITLE
Upgrade Typescript from 4.7.4 to 4.9.4 to be able to use satisfies keyword

### ISSUE THREAD

**Issue by SirMishaa** [vuejs/create-vue#215]
Hello,
I like this template a lot, that's a good work

I'd like to use the [satisfies keyword](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html) in my project. 

I just created a new project, but it seems to use an older version of Typescript, could we upgrade to the latest? I can do a PR for that. 
```bash
❯ yarn upgrade-interactive --latest
yarn upgrade-interactive v1.22.19
info Color legend :
 "<red>"    : Major Update backward-incompatible updates
 "<yellow>" : Minor Update backward-compatible features
 "<green>"  : Patch Update backward-compatible bug fixes
? Choose which packages to update. (Press <space> to select, <a> to toggle all, <i> to invert selection)
 devDependencies
   name        range   from      to     url
❯◯ typescript  latest  4.7.4  ❯  4.9.4  https://www.typescriptlang.org/
```

Kind regards

**Comment by SirMishaa** [comment]
This is a bit the same for Vite, would be nice if we could upgrade to Vite 4 

Let me know what you think about it. I'd be glad to help if needed

**Comment by LinusBorg** [comment]
FYI: TS 4.9 has an internal bug that breaks import-autocompletes in SFCs with Volar. See:

https://github.com/johnsoncodehk/volar/issues/2190

I think we should upgrade only to 4.8 until that is resolved.

**Comment by FedorSherbakov** [comment]
Hello 👋

Looks like it's resolved now https://github.com/vuejs/language-tools/issues/2190#issuecomment-1482169494

**Comment by SirMishaa** [comment]
Should we upgrade to the latest stable release, since it's fixed ? The latest version is 5.0.2

**Comment by sodatea** [comment]
Upgraded in the latest release.
